### PR TITLE
feat: added consumer and tests

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #1291 
Added consumer to insert and remove records from database, provided queue to adapters and added initiation of scheduler.

Also added tests. 

Note, that consumer accepts connection, not data source. Such approach is good for [sqlite db](https://stackoverflow.com/questions/15822778/sqlite-connection-pool-in-java-locked-database), but will not be suitable for connection pool and normal db (as postgresql or mysql).

### CODE REVIEW

**Comment by dgarus** [comment]
> but will not be suitable for connection pool and normal db (as postgresql or mysql)

Why not?

**Comment by olenagerasimova** [comment]
@dgarus please, check

**Review by ChGen — APPROVED** [review]
Looks ok, thanks!

File: src/main/java/com/artipie/db/DbConsumer.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                        }
+                    } catch (final SQLException ex) {
+                        Logger.error(this, ex.getMessage());
+                        errors.add(record);
+                    }
+                }
+                conn.commit();
+            } catch (final SQLException ex) {
+                Logger.error(this, ex.getMessage());
+                events.forEach(DbConsumer.this.subject::onNext);
```
Comment (dgarus) [inline src/main/java/com/artipie/db/DbConsumer.java:116]: Could you please explain this line?

File: src/main/java/com/artipie/db/DbConsumer.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                error = true;
+            }
+            if (!error) {
+                errors.forEach(DbConsumer.this.subject::onNext);
```
Comment (dgarus) [inline src/main/java/com/artipie/db/DbConsumer.java:120]: Could you please explain this line?

**Comment by olenagerasimova** [comment]
please, do not merge. I'm figuring out what is wrong with smoke-tests

**Review by dgarus — DISMISSED** [review]
lgtm

**Review by ChGen — APPROVED** [review]
LGTM

**Pull request merged**